### PR TITLE
Connect package

### PR DESCRIPTION
**Pull request by blackout1471**

### CODE REVIEW

File: Source/Protocol/Validators/Rules/GenerateSessionMessageRule.h [hunk 1]
```diff
@@ -9,12 +9,12 @@
 namespace MQTT {
 	namespace Protocol {
 		namespace Validators {
-			class GenerateSessionMessageRule : public IRule
+			class CanClientContinueSession : public IRule
```
Comment (Ripiter) [inline Source/Protocol/Validators/Rules/GenerateSessionMessageRule.h:12]: Change the file name

File: Source/Server/MqttService.cpp [hunk 2]
```diff
@@ -165,12 +137,12 @@ namespace MQTT {
...
 		}
 
-
-		void MqttService::OnClientDisconnect(const Client& client, const Protocol::DisconnectPackage& package)
+		std::string MqttService::GenerateUniqueId()
 		{
-			//TODO: Remove will message when storage of it is implemented.
-			DisconnectClientState(client);
+			// TODO: Create unique id
+			return "1";
```
Comment (Ripiter) [inline Source/Server/MqttService.cpp]: Use the generate uid from client utility

File: Source/Server/MqttService.h [hunk 3]
```diff
@@ -25,6 +26,7 @@ namespace MQTT {
 
 
 			MqttClient* GetClientState(const std::string& clientId);
+			std::string GenerateUniqueId();
```
Comment (Ripiter) [inline Source/Server/MqttService.h]: Remove from header file

Comment (Geefy) [inline Source/Server/MqttService.h]: Skal fjernes fra klassen da den ikke længere er i brug

File: Source/Server/MqttService.cpp [hunk 4]
```diff
@@ -165,12 +137,12 @@ namespace MQTT {
 			return nullptr;
 		}
 
-
-		void MqttService::OnClientDisconnect(const Client& client, const Protocol::DisconnectPackage& package)
+		std::string MqttService::GenerateUniqueId()
```
Comment (Geefy) [inline Source/Server/MqttService.cpp]: Skal fjernes da den ikke længere bruges

File: Source/Server/MqttService.cpp [hunk 5]
```diff
@@ -103,57 +84,48 @@ namespace MQTT {
...
+		void MqttService::OnClientDisconnect(const Client& client, const Protocol::DisconnectPackage& package)
+		{
+			//TODO: Remove will message when storage of it is implemented.
+			DisconnectClientState(client);
+		}
 
-			if (shouldContinueSession)
+		void MqttService::DisconnectClientState(const Client& client)
+		{
+			for (auto& arr_client : m_ClientStates)
```
Comment (Geefy) [inline Source/Server/MqttService.cpp]: arr_client -> arr_Client?

**Review by Geefy — CHANGES_REQUESTED** [review]

File: Source/Server/MqttService.cpp [hunk 6]
```diff
@@ -103,57 +84,48 @@ namespace MQTT {
...
-			}).Run());
+			auto action = Protocol::Validators::ConnectValidator()
+				.ValidateClient(package, m_ClientStates, clientState);
+
+			clientState->ConnectionIdentifier = client.GetIdentifier();
 
-			if (shouldDisconnectClient)
+			switch (action)
 			{
+			case MQTT::Protocol::Validators::ConnectValidator::Disconnect:
```
Comment (Ripiter) [inline Source/Server/MqttService.cpp:93]: should we set client state to disconnected if he is connected already?

File: Source/Server/MqttService.cpp [hunk 7]
```diff
@@ -103,57 +84,48 @@ namespace MQTT {
...
+				clientState->IsConnected = true;
+				break;
 			}
+		}
 
-			std::vector<unsigned char> message;
-			auto shouldContinueSession = (RuleEngine({
-				{new ContinueSessionRule(package.VariableHeader.VariableLevel, packageClientId, m_ClientStates), true}
-			}).Run());
+		void MqttService::OnClientDisconnect(const Client& client, const Protocol::DisconnectPackage& package)
```
Comment (Ripiter) [inline Source/Server/MqttService.cpp]: package is not used

File: Source/Protocol/Validators/Rules/GenerateSessionMessageRule.h (hunk 1 above)
Comment (blackout1471) [reply Source/Protocol/Validators/Rules/GenerateSessionMessageRule.h:12]: Fixed, changes name

File: Source/Server/MqttService.cpp (hunk 2 above)
Comment (blackout1471) [reply Source/Server/MqttService.cpp]: Resolved by deleting method. Not used anymore

File: Source/Server/MqttService.h (hunk 3 above)
Comment (blackout1471) [reply Source/Server/MqttService.h]: Resolved by deleting method. Not used anymore

Comment (blackout1471) [reply Source/Server/MqttService.h]: Resolved by deleting method. Not used anymore

File: Source/Server/MqttService.cpp (hunk 4 above)
Comment (blackout1471) [reply Source/Server/MqttService.cpp]: Resolved by deleting method. Not used anymore

File: Source/Server/MqttService.cpp (hunk 5 above)
Comment (blackout1471) [reply Source/Server/MqttService.cpp]: Renamed to clientState: (clientState : m_ClientStates)

File: Source/Server/MqttService.cpp (hunk 6 above)
Comment (blackout1471) [reply Source/Server/MqttService.cpp:93]: Yes, this is the action after the validation. So if the package is malformed, we have to disconnect the client from the current session.

File: Source/Server/MqttService.cpp (hunk 7 above)
Comment (blackout1471) [reply Source/Server/MqttService.cpp]: Removed the package from the argument list.

**Review by Geefy — APPROVED** [review]

**Pull request merged**